packages/slice-machine/src/modules/screenshots/sagas.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  call,
+  fork,
+  takeLatest,
+  put,
+  SagaReturnType,
+} from "redux-saga/effects";
+import { getType } from "typesafe-actions";
+import { withLoader } from "../loading";
+import { LoadingKeysEnum } from "../loading/types";
+import {
+  generateSliceScreenshotCreator,
+  generateSliceCustomScreenshotCreator,
+} from "./actions";
+import {
+  generateSliceScreenshotApiClient,
+  generateSliceCustomScreenshotApiClient,
+} from "@src/apiClient";
+import { openToasterCreator, ToasterType } from "@src/modules/toaster";
+
+export function* generateSliceScreenshotSaga({
+  payload,
+}: ReturnType<typeof generateSliceScreenshotCreator.request>) {
+  const { component, variationId, setData } = payload;
+  try {
+    setData({
```

Comment:
i don't think this setData should be needed anymore on the new page.
We either display toasts in case of error or success and we have global loading states.